lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -221,28 +221,23 @@ SmallVector<unsigned> getContigPerThread(Attribute layout) {
   }
 }
 
-SmallVector<unsigned> getUniqueContigPerThread(Type type) {
-  if (type.isIntOrIndexOrFloat() || type.isa<triton::PointerType>())
-    return SmallVector<unsigned>(1, 1);
-  auto tensorType = type.cast<RankedTensorType>();
-  auto shape = tensorType.getShape();
+SmallVector<unsigned> getUniqueContigPerThread(Attribute layout,
```

Comment:
Could it stay with the same input argument list? i.e., `getUniqueContigPerThread(Type type)`. I think `std::min(srcReduceDimSize / elementPerThreads,` is the only problem here, right?

Reply:
Right I could make it work by creating a RankedTensorType with the layout and shape there but it seemed a little less clean. Also this way it has same signature as `getThreadsPerWarpWithUniqueData` and other functions. Do you think the original interface was better? I can definitely leave it like that if you prefer.

Reply:
It's OK, I found that we already have `getThreadsPerWarpWithUniqueData` with two input arguments.